Engine_Revit_UI/Modify/SetParameter.cs

```diff
@@ -133,8 +133,25 @@ public static bool SetParameter(this Element element, BuiltInParameter builtInPa
 
         public static bool SetParameter(this Parameter parameter, object value, Document document = null)
         {
+            // Skip null and read-only parameters
             if (parameter == null || parameter.IsReadOnly)
                 return false;
+            
+            // Workset parameters
+            // Filter for worksets
+            if (parameter.Id.IntegerValue == (int)BuiltInParameter.ELEM_PARTITION_PARAM)
+            {
+                // Skip non-workshared documents because they can't contain worksets
+                if (!document.IsWorkshared)
+                    return false;
+
+                // Create a new workset with a specified name or return an existing one if it already exists
+                string worksetName = value as string;
+                Workset workset = Create.Workset(document, worksetName);
+
+                // Set the "Workset" parameter to the newly created workset
+                return parameter.Set(workset.Id.IntegerValue);
```

Comment:
I would add an extra check here - by the read of code this may crash in case of whitespace workset name being fed in.

```suggestion
if (workset != null)
    return parameter.Set(workset.Id.IntegerValue);
else
    return false;
```

Reply:
Empty workset names are already checked for in the Create method and return a null.
Would a null check like this suffice?
```
if (workset != null)
    return parameter.Set(workset.Id.IntegerValue);
```

Reply:
Not really, because then the code will continue an empty run, even though we already know the outcome of this method will be `false`.

Reply:
A final 'else' statement will be added to capture that and return a false